controller/src/main/java/io/pravega/controller/server/ControllerServiceStarter.java

```diff
@@ -353,6 +360,12 @@ protected void shutDown() throws Exception {
             log.info("Closing storeClient");
             storeClient.close();
 
+            // Close metrics.
+            if (streamMetrics != null) {
+                streamMetrics.close();
+                transactionMetrics.close();
```

Comment:
should we not check for transactionMetrics to be null

Reply:
Done.